src/js/components/Carousel/__tests__/Carousel-test.tsx

```diff
@@ -1,139 +1,249 @@
+  test('selector navigation: forward', async () => {
+    render(
+      <Grommet>
+        <Carousel controls="selectors">
+          <div>Slide One</div>
+          <div>Slide Two</div>
+          <div>Slide Three</div>
+        </Carousel>
+      </Grommet>,
+    );
 
-    fireEvent.focus(button!);
-    expect(onFocus).toHaveBeenCalledTimes(1);
+    /**
+     * - Currently on "Slide One"
+     * - Simulating click on the 3rd selector button
+     * - Expecting "Slide Three" to be visible
+     */
+    const thirdSelector = screen.getByLabelText('Jump to slide 3');
+    userEvent.click(thirdSelector);
+    await waitFor(() => {
+      expect(getSlideOne()).not.toBeVisible();
+      expect(getSlideThree()).toBeVisible();
+    });
+  });
 
-    fireEvent.click(button!);
-    expect(container.firstChild).toMatchSnapshot();
+  test('selector navigation: backward', async () => {
+    render(
+      <Grommet>
+        <Carousel initialChild={2} controls="selectors">
+          <div>Slide One</div>
+          <div>Slide Two</div>
+          <div>Slide Three</div>
+        </Carousel>
+      </Grommet>,
+    );
 
-    fireEvent.blur(button!);
-    expect(onBlur).toHaveBeenCalledTimes(1);
+    /**
+     * - Currently on "Slide Three"
+     * - Simulating click on the 1st selector button
+     * - Expecting "Slide One" to be visible
+     */
+    const firstSelector = screen.getByLabelText('Jump to slide 1');
+    userEvent.click(firstSelector);
+    await waitFor(() => {
+      expect(getSlideThree()).not.toBeVisible();
+      expect(getSlideOne()).toBeVisible();
+    });
   });
 
-  test('play', () => {
+  test('play', async () => {
     jest.useFakeTimers('modern');
-    const { container } = render(
+    render(
       <Grommet>
-        <Carousel play={1000}>
-          <Image src="//v2.grommet.io/assets/IMG_4245.jpg" />
-          <Image src="//v2.grommet.io/assets/IMG_4210.jpg" />
+        <Carousel controls={false} play={500}>
+          <div>Slide One</div>
+          <div>Slide Two</div>
         </Carousel>
       </Grommet>,
     );
-    expect(container.firstChild).toMatchSnapshot();
-    // Advance timers so the carousel advances
-    act(() => {
-      jest.advanceTimersByTime(1300);
+
+    expect(getSlideOne()).toBeVisible();
+    await act(async () => {
+      jest.advanceTimersByTime(1200);
+    });
+    await waitFor(() => {
+      expect(getSlideOne()).not.toBeVisible();
+      expect(getSlideTwo()).toBeVisible();
     });
-    expect(container.firstChild).toMatchSnapshot();
   });
 
-  test('should work as controlled carousel', () => {
-    const setActiveSlide = jest.fn();
+  test('keyboard: left arrow', async () => {
+    render(
+      <Grommet>
+        <Carousel initialChild={1} a11yTitle="test-carousel">
+          <div>Slide One</div>
+          <div>Slide Two</div>
+        </Carousel>
+      </Grommet>,
+    );
+    const carousel = screen.getByLabelText('test-carousel');
+    fireEvent.keyDown(carousel, {
```

Comment:
Can we use `userEvent` here instead of `fireEvent`?

Reply:
This is the one case where you wanna use `fireEvent` instead of `userEvent` https://testing-library.com/docs/guide-events/#alternatives